netbox/netbox/api/authentication.py

```diff
@@ -123,6 +134,33 @@ def has_object_permission(self, request, view, obj):
 
         return super().has_object_permission(request, view, obj)
 
+    # Helper function to obtain the desired action from the first permission in the permission map
+    def get_action(self, method):
+        # Get the first permission
+        permission = self.perms_map.get(method)[0] if len(self.perms_map.get(method)) > 0 else None
+        if permission:
+            # Remove app and model label
+            action = permission.replace('%(app_label)s.', '').replace('_%(model_name)s', '')
+            return action
+        elif action := HTTP_ACTIONS[method]:
+            return action
+        return None
+
+
+class ViewOnlyPermissions(TokenPermissions):
+    """
+    Override the stock perm_map to require only view permissions
+    """
+    perms_map = {
```

Comment:
Rather than overriding `perms_map`, it might be cleaner to just override `get_action()` to return `'%(app_label)s.view_%(model_name)s'` if the request method is not `OPTIONS`.